File: 8kyu/(8 kyu) Opposites Attract/(8 kyu) Opposites Attract.cs
using System;

// 1 Plain solution
public class LoveDetector
{
    public static bool lovefunc(int flower1, int flower2)
    {
        bool positiveCase1 = (flower1 % 2 == 0) && (flower2 % 2 != 0);
        bool positiveCase2 = (flower1 % 2 != 0) && (flower2 % 2 == 0);

        if (positiveCase1 || positiveCase2) {
            return true;
        } else {
            return false;
        }
    }
}

// 2 Optimized solution
public class LoveDetector
{
    public static bool lovefunc(int flower1, int flower2)
    {
        bool isF1Even = (flower1 % 2 == 0);
        bool isF2Even = (flower2 % 2 == 0);
        return isF1Even != isF2Even;
    }
}

// 3 Clever solution
public class LoveDetector
{
    public static bool lovefunc(int flower1, int flower2)
    {
        return (flower1 + flower2) % 2 == 1;
    }
}

// 4 Coding golf
public class LoveDetector
{
    public static bool lovefunc(int f1,int f2)=>(f1+f2)%2==1;
}

File: 8kyu/(8 kyu) Opposites Attract/(8 kyu) Opposites Attract.py
# 1 Simple solution
def lovefunc(flower1, flower2):
    positiveCase1 = flower1 % 2 == 0 and flower2 % 2 != 0
    positiveCase2 = flower1 % 2 != 0 and flower2 % 2 == 0
    res = False

    if positiveCase1 or positiveCase2:
        res = True

    return res


# 2 Optimized solution
def lovefunc(flower1, flower2):
    positiveCase1 = flower1 % 2 == 0 and flower2 % 2 != 0
    positiveCase2 = flower1 % 2 != 0 and flower2 % 2 == 0

    return positiveCase1 or positiveCase2


# 3 Clever solution
def lovefunc(flower1, flower2):
    return flower1 % 2 != flower2 % 2

# 4 Coding golf
lovefunc=lambda f1,f2:f1%2!=f2%2
